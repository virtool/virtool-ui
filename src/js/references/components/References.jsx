import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { mapSettingsStateToProps } from "../../administration/mappers";
import { Container, LoadingPlaceholder, ContainerNarrow, ViewHeader, ViewHeaderTitle } from "../../base";
import SourceTypes from "./SourceTypes/SourceTypes";
import ReferenceList from "./List";
import ReferenceDetail from "./Detail/Detail";
import AddReference from "./Add";

export const ReferenceSettings = () => (
    <ContainerNarrow>
        <ViewHeader title="Reference Settings">
            <ViewHeaderTitle>Settings</ViewHeaderTitle>
        </ViewHeader>
        <SourceTypes global />
    </ContainerNarrow>
);

export const References = ({ loading }) => {
    if (loading) {
        return <LoadingPlaceholder />;
    }

    return (
        <Container>
            <Switch>
                <Route path="/refs" component={ReferenceList} exact />
                <Redirect from="/refs/settings/*" to="/refs/settings" />
                <Route path="/refs/settings" component={ReferenceSettings} />
                <Route path="/refs/add" component={AddReference} />
                <Route path="/refs/:refId" component={ReferenceDetail} />
            </Switch>
        </Container>
    );
};

export default connect(mapSettingsStateToProps)(References);
