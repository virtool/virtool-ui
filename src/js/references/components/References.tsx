import { useFetchSettings } from "@administration/queries";
import { Container, LoadingPlaceholder } from "@base";
import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import ReferenceDetail from "./Detail/ReferenceDetail";
import ReferenceList from "./ReferenceList";
import { ReferenceSettings } from "./ReferenceSettings";

/**
 * The references view with routes to reference-related components
 */
export default function References() {
    const { isPending } = useFetchSettings();

    if (isPending) {
        return <LoadingPlaceholder />;
    }

    return (
        <Container>
            <Switch>
                <Route path="/refs" component={ReferenceList} exact />
                <Redirect from="/refs/settings/*" to="/refs/settings" />
                <Route path="/refs/settings" component={ReferenceSettings} />
                <Route path="/refs/:refId" component={ReferenceDetail} />
            </Switch>
        </Container>
    );
}
