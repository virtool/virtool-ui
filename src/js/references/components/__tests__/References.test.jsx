import { screen } from "@testing-library/react";
import nock from "nock";
import React from "react";
import { test } from "vitest";
import { renderWithProviders } from "../../../../tests/setupTests";
import { References } from "../References";

test("<References />", () => {
    nock("http://localhost").get("/api/settings").reply(200, {});

    renderWithProviders(<References />);

    expect(screen.queryByText("Loading")).not.toBeInTheDocument();

    it.each([true, false])("should render when [loading=%p]", loading => {
        const wrapper = shallow(<References loading={loading} />);
        expect(wrapper).toMatchSnapshot();
    });
});
