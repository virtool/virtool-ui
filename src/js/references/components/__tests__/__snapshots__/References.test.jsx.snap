// Vitest Snapshot v1

exports[`<ReferenceSettings /> > should render 1`] = `
<ContainerNarrow>
  <ViewHeader
    title="Reference Settings"
  >
    <ViewHeaderTitle>
      Settings
    </ViewHeaderTitle>
  </ViewHeader>
  <Connect(SourceTypes)
    global={true}
  />
</ContainerNarrow>
`;

exports[`<References /> > should render when [loading=%p] false 1`] = `
<Container>
  <Switch>
    <Route
      component={
        {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/refs"
    />
    <Redirect
      from="/refs/settings/*"
      to="/refs/settings"
    />
    <Route
      component={[Function]}
      path="/refs/settings"
    />
    <Route
      component={
        {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/refs/add"
    />
    <Route
      component={
        {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      path="/refs/:refId"
    />
  </Switch>
</Container>
`;

exports[`<References /> > should render when [loading=%p] true 1`] = `<LoadingPlaceholder />`;
